components/places/src/api/matcher.rs

```diff
@@ -691,4 +696,50 @@ mod tests {
             }]
         );
     }
+    #[test]
+    fn search_unicode() {
+        let mut conn = PlacesDb::open_in_memory(None).expect("no memory db");
+
+        let url = Url::parse("http://exämple.com/123").unwrap();
+        let visit = VisitObservation::new(url.clone())
+            .with_title("Example page 123".to_string())
+            .with_visit_type(VisitTransition::Typed)
+            .with_at(Timestamp::now());
+
+        apply_observation(&mut conn, visit).expect("Should apply visit");
+
+        let by_url_without_path = search_frecent(
+            &conn,
+            SearchParams {
+                search_string: "http://exämple.com".into(),
+                limit: 10,
+            },
+        )
+        .expect("Should search by URL without path");
+        assert!(by_url_without_path
+            .iter()
+            // Should we consider un-punycoding the title? (firefox desktop doesn't...)
```

Comment:
I think it's worth considering. If callers don't need to Punycode on the way in (I _think_ the history tracking machinery takes care of that?), it makes sense to decode on the way out in autocomplete.

On the other hand, if they want to do any string slicing or other fancy things with the title, maybe they should also decode it, to minimize the chance of breakage because "it works for ASCII, so good enough". (But, on the other, other hand, that assumption will also likely break for Punycoded results, so 🤷‍♀️)

Let's discuss in a follow-up issue.

Reply:
The other downside is that sometimes displaying the real url is done deliberately to prevent confusion.